dags/common/pull_ftp.py

```diff
@@ -40,13 +40,20 @@ def migrate_from_ftp(
     sftp: SFTPService, repo: IRepository, logger: PrintLogger, **kwargs
 ):
     params = kwargs["params"]
-    if "force_pull" in params and params["force_pull"]:
-        return _force_pull(sftp, repo, logger, **kwargs)
-    elif (
+    force_pull_specific_files = (
         "filenames_pull" in params
         and params["filenames_pull"]["enabled"]
         and params["filenames_pull"]["force_from_ftp"]
-    ):
+    )
+    force_pull_all_files = (
```

Comment:
I am just thinking, do you really need `force_pull` param? Is it happening only when you add `excluded_directories`?

Reply:
no, let me explain:
* `force_pull` is true, will force-pull all files in the SFTP.
* `force_pull` is false, just the missing files are pulled.
* If there is at least one dir in the excluded_directories, this dir will be ignored, but files from all files which are missing from the rest of the folder will be downloaded to S3. 
* For example, we have 3 folder in sftp: folder1, folder2, and folder3.

`force_pull` is **false**. 
`exluded_dir` is ["folder1"] 
In this case, missing files just from folder2 and folder3 will be downloaded. 

`force_pull` is **false**
If the `excluded_dirs` is []
The missing files from folder1, folder2, and folder3 will be downloaded again.

`force_pull` is **true**. 
`exluded_dir` is ["folder1"] 
In this case, ALL files just from folder2 and folder3 will be downloaded again (and will rewrite the files in S3 if they were already downloaded). 

`force_pull` is **false**
If the `excluded_dirs` is []
ALL files from folder1, folder2, and folder3 will be downloaded again (and will rewrite the files in S3 if they were already downloaded).

Reply:
thanks for the explenation :)